lib/src/controller/database.dart

```diff
@@ -1,13 +1,504 @@
+  // Display
+  TextColumn get appLanguage => text().nullable()();
+  TextColumn get themeMode => textEnum<ThemeMode>().nullable()();
+  TextColumn get colorScheme => textEnum<FlexScheme>().nullable()();
+  BoolColumn get enableTrueBlack => boolean().nullable()();
+  BoolColumn get compactMode => boolean().nullable()();
+  BoolColumn get hideActionButtons => boolean().nullable()();
+  BoolColumn get hideFeedUIOnScroll => boolean().nullable()();
+  RealColumn get globalTextScale => real().nullable()();
+  BoolColumn get alwaysShowInstance => boolean().nullable()();
+  BoolColumn get coverMediaMarkedSensitive => boolean().nullable()();
+  BoolColumn get fullImageSizeThreads => boolean().nullable()();
+  BoolColumn get fullImageSizeMicroblogs => boolean().nullable()();
+  // Feed defaults
+  TextColumn get feedDefaultView => textEnum<FeedView>().nullable()();
+  TextColumn get feedDefaultFilter => textEnum<FeedSource>().nullable()();
+  TextColumn get feedDefaultThreadsSort => textEnum<FeedSort>().nullable()();
+  TextColumn get feedDefaultMicroblogSort => textEnum<FeedSort>().nullable()();
+  TextColumn get feedDefaultCombinedSort => textEnum<FeedSort>().nullable()();
+  TextColumn get feedDefaultExploreSort => textEnum<FeedSort>().nullable()();
+  TextColumn get feedDefaultCommentSort => textEnum<CommentSort>().nullable()();
+  BoolColumn get feedDefaultHideReadPosts => boolean().nullable()();
+  // Feed actions
+  TextColumn get feedActionBackToTop => textEnum<ActionLocation>().nullable()();
+  TextColumn get feedActionCreateNew => textEnum<ActionLocation>().nullable()();
+  TextColumn get feedActionExpandFab => textEnum<ActionLocation>().nullable()();
+  TextColumn get feedActionRefresh => textEnum<ActionLocation>().nullable()();
+  TextColumn get feedActionSetFilter =>
+      textEnum<ActionLocationWithTabs>().nullable()();
+  TextColumn get feedActionSetSort => textEnum<ActionLocation>().nullable()();
+  TextColumn get feedActionSetView =>
+      textEnum<ActionLocationWithTabs>().nullable()();
+  TextColumn get feedActionHideReadPosts =>
+      textEnum<ActionLocation>().nullable()();
+  // Swipe actions
+  BoolColumn get enableSwipeActions => boolean().nullable()();
+  TextColumn get swipeActionLeftShort => textEnum<SwipeAction>().nullable()();
+  TextColumn get swipeActionLeftLong => textEnum<SwipeAction>().nullable()();
+  TextColumn get swipeActionRightShort => textEnum<SwipeAction>().nullable()();
+  TextColumn get swipeActionRightLong => textEnum<SwipeAction>().nullable()();
+  RealColumn get swipeActionThreshold => real().nullable()();
+  // Filter list activations
+  TextColumn get filterLists =>
+      text().nullable().map(const FilterListActivationConverter())();
+  BoolColumn get showErrors => boolean().nullable()();
+
+  @override
+  Set<Column<Object>> get primaryKey => {name};
+}
+
+// Table to store misc data that is either one off or doesn't fit into other tables
+class MiscCache extends Table {
```

Comment:
A single row table might be preferable to using key-value pairs. I came across [this discussion](https://stackoverflow.com/questions/2300356/using-a-single-row-configuration-table-in-sql-server-database-bad-idea/2300493#2300493) comparing the two techniques. One thing that could make a single row table useful for us is the ability to set up foreign key constraints (e.g., with selected account, selected profile, or auto switch profile).